include/llama.h

```diff
@@ -210,6 +210,14 @@ extern "C" {
         bool sorted;      // note: do not assume the data is sorted - always check this flag
     } llama_token_data_array;
 
+    struct llama_sampler_ggml_data {
+        struct ggml_tensor * logits;
+        struct ggml_tensor * probs;
+        struct ggml_tensor * sampled_token;
+        struct ggml_tensor * filtered_ids;
+        int32_t              batch_idx;
```

Comment:
`batch_idx` is unused - do you plan to use it for anything?

Reply:
I was planning on using it with the penalties sampler but it turned out not to be needed. I'll remove it shortly.